### ISSUE TITLE
windows troubles with test_results filenames

### ISSUE THREAD

**Issue by frankpuppa** [avocado-framework/avocado#2189]
Hello guys, I noticed that there is a little problem when the directories of the test results are moved from one OS to another. The problem I noticed arises mainly on window which has stricter filename rules than our friend Linux. 
In the example I'm giving, the directories inside the test_results folder take a name as follow:
<Filename.py>:<TestName>;<variant>
here is an example of one of my test:
1-testlocal_test_detect_local.py:TestMmc.test;sda-341b
When these folders are moved on Windows machines, windows replaces every colon of the filename with another character. This cause a little issue in the link names contained inside the result.html/.xml files.

**Comment by apahim** [comment]
According to https://msdn.microsoft.com/en-us/library/aa365247:
```
The following reserved characters:

    < (less than)
    > (greater than)
    : (colon)
    " (double quote)
    / (forward slash)
    \ (backslash)
    | (vertical bar or pipe)
    ? (question mark)
    * (asterisk)
```
Should we change our Test ID format (https://www.redhat.com/archives/avocado-devel/2016-March/msg00024.html) to make Windows happy?

**Comment by ldoktor** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Test-name issue on Window](https://trello.com/c/iPZihxhn/1120-test-name-issue-on-window)

**Comment by frankpuppa** [comment]
Hi, the problem is affected by the colon in the filename, not the semi-colon, as you have instead pointed in here(https://trello.com/c/iPZihxhn/1120-test-name-issue-on-windowl). I believe it was a typo, but it is better to avoid confusion..

**Comment by lmr** [comment]
My choice would be to change the format. Windows isn't going away any time soon so making it play nice with that OS is a good idea moving forward.

**Comment by ldoktor** [comment]
Well actually it's not just the name, it's also about the length as some systems allow longer file-names, others don't. On windows that includes the full path and not just the file name (if I understand the https://msdn.microsoft.com/en-us/library/aa365247#maxpath correctly).

Anyway, yes, we definitely want to be able to display results on windows so we'll have to find a suitable way for both, human and computers).

**Comment by adereis** [comment]
I added a comment there: https://trello.com/c/iPZihxhn/1120-serialized-test-ids-contain-the-character-invalid-in-windows#comment-59b98ae84ca2f03eca0cfa16

**Comment by clebergnu** [comment]
The issue reported here has been fixed on PR #2201.

**Comment by clebergnu** [comment]
Fixed on #2233
